lib/Doctrine/ORM/Cache/Persister/Collection/AbstractCollectionPersister.php

```diff
@@ -30,10 +30,10 @@
 use Doctrine\ORM\Cache\Region;
 use Doctrine\ORM\EntityManagerInterface;
 use Doctrine\ORM\Mapping\ClassMetadata;
-use Doctrine\ORM\Mapping\ClassMetadataFactory;
 use Doctrine\ORM\PersistentCollection;
 use Doctrine\ORM\Persisters\Collection\CollectionPersister;
 use Doctrine\ORM\UnitOfWork;
+use Doctrine\Persistence\Mapping\ClassMetadataFactory;
```

Comment:
This looks wrong… maybe the right fix here would be to assert that the value we are assigning is an instance of the ORM CMF… or can `$em->getMetadataFactory();` really return something else?

Reply:
I had the same doubt here. I don't know ORM's internals good enough to know if this is right or wrong.

Reply:
Then please revert that one.

Reply:
I found a good solution that just involves adding a comment to `EntityManagerInterface`, I'm going to push it to your branch soon